src/App.js

```diff
@@ -984,18 +987,21 @@ export class App extends React.Component<AppProps, AppState> {
             const serializedProgram = this.programSerializer.serialize(this.state.programSequence.getProgram());
             const serializedCharacterState = this.characterStateSerializer.serialize(this.state.characterState);
             const serializedAllowedActions = this.allowedActionsSerializer.serialize(this.state.allowedActions);
-            window.history.pushState(
-                {
-                    p: serializedProgram,
-                    c: serializedCharacterState,
-                    t: this.state.settings.theme,
-                    a: serializedAllowedActions,
-                    w: this.state.settings.world
-                },
-                '',
-                Utils.generateEncodedProgramURL(this.version, this.state.settings.theme, this.state.settings.world, serializedProgram, serializedCharacterState, serializedAllowedActions),
-                '',
-            );
+            clearTimeout(this.pushStateTimer);
+            this.pushStateTimer = setTimeout(() => {
+                window.history.pushState(
+                    {
+                        p: serializedProgram,
+                        c: serializedCharacterState,
+                        t: this.state.settings.theme,
+                        a: serializedAllowedActions,
+                        w: this.state.settings.world
+                    },
+                    '',
+                    Utils.generateEncodedProgramURL(this.version, this.state.settings.theme, this.state.settings.world, serializedProgram, serializedCharacterState, serializedAllowedActions),
+                    '',
+                );
+            }, 300);
```

Comment:
I think we should increase the value of the delay by a little bit, as 300ms is right at the limit of 100 times in 30 seconds. Say to 350ms. I think that it would also be worthwhile to make a `const` variable for this value so that we can give it a name: `pushStateDelayMs`.